b2sdk/account_info/sqlite_account_info.py

```diff
@@ -34,17 +34,28 @@ class SqliteAccountInfo(UrlPoolAccountInfo):
     completed.
     """
 
-    def __init__(self, file_name=None, last_upgrade_to_run=None):
+    def __init__(self, file_name=None, env=os.environ, last_upgrade_to_run=None):
         """
-        If ``file_name`` argument is empty or ``None``, path from ``B2_ACCOUNT_INFO`` environment variable is used. If that is not available, a default of ``~/.b2_account_info`` is used.
+        If ``file_name`` argument is empty or ``None``, the path from
+        ``B2_ACCOUNT_INFO`` environment variable is used.  If that is not
+        available, we check if the default ``~/.b2_account_info`` exists to
+        ensure backwards compatibility, before using ``XDG_CONFIG_HOME``. If
+        ``XDG_CONFIG_HOME`` is not set, we fall back to the default.
 
         :param str file_name: The sqlite file to use; overrides the default.
+        :param dict env: Override Environment variables. For testing only.
         :param int last_upgrade_to_run: For testing only, override the auto-update on the db.
         """
         self.thread_local = threading.local()
-        user_account_info_path = file_name or os.environ.get(
-            B2_ACCOUNT_INFO_ENV_VAR, B2_ACCOUNT_INFO_DEFAULT_FILE
-        )
+        if B2_ACCOUNT_INFO_ENV_VAR in env:
```

Comment:
I think this code works, but the logic would be clearer if it mirrored the structure of the comment, above:

```
if file_name:
    use it
elif B2_ACCOUNT_INFO_ENV_VAR in env:
    use it
elif default file exists:
    use it
elif XDG_CONFIG_HOME in env:
    use it
else:
    use default path
```